src/devices/Mcp9808/Mcp9808.cs

```diff
@@ -0,0 +1,206 @@
+            }
+        }
+
+        /// <summary>
+        /// Checks if the device is a MCP9808
+        /// </summary>
+        /// <returns>True if device has been correctly detected</returns>
+        private bool Init()
+        {
+            if (Read16(Register8.MCP_MANUF_ID) != 0x0054)
+            {
+                return false;
+            }
+
+            if (Read16(Register8.MCP_DEVICE_ID) != 0x0400)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Return the internal resolution register
+        /// </summary>
+        /// <returns>Resolution setting</returns>
+        public byte GetResolution()
+        {
+            return Read8(Register8.MCP_RESOLUTION);
+        }
+
+        /// <summary>
+        /// Wakes-up the device
+        /// </summary>
+        public void Wake()
+        {
+            SetShutdown(false);
+
+            // Sleep 250 ms, which is the typical temperature conversion time for the highest resolution (page 3 in datasheet, tCONV)
+            Thread.Sleep(250);
+        }
+
+        /// <summary>
+        /// Shuts down the device
+        /// </summary>
+        public void Shutdown()
+        {
+            SetShutdown(true);
+        }
+
+        /// <summary>
+        /// Read MCP9808 Temperature (℃)
+        /// </summary>
+        /// <returns>Temperature</returns>
+        private double GetTemperature()
+        {
+            ushort value = Read16(Register8.MCP_AMBIENT_TEMP);
+
+            if (value == 0xFFFF)
+            {
+                return double.NaN;
+            }
+
+            double temp = value & 0x0FFF;
+            temp /= 16.0;
+            if ((value & 0x1000) != 0)
+            {
+                temp -= 256;
+            }
+
+            return Math.Round(temp, 5);
+        }
+
+        /// <summary>
+        /// Set MCP9808 Shutdown
+        /// </summary>
+        /// <param name="isShutdown">Shutdown when value is true.</param>
+        private void SetShutdown(bool isShutdown)
+        {
+            Register16 curVal = ReadRegister16(Register8.MCP_CONFIG);
+
+            if (isShutdown)
+            {
+                curVal |= Register16.MCP_CONFIG_SHUTDOWN;
+            }
+            else
+            {
+                curVal &= ~Register16.MCP_CONFIG_SHUTDOWN;
+            }
+
+            Write16(Register8.MCP_CONFIG, curVal);
+        }
+
+        /// <summary>
+        /// Cleanup
+        /// </summary>
+        public void Dispose()
+        {
+            _i2cDevice?.Dispose();
+            _i2cDevice = null;
+        }
+
+        internal Register16 ReadRegister16(Register8 reg)
+        {
+            return (Register16)Read16(reg);
+        }
+
+        internal ushort Read16(Register8 reg)
+        {
+            _i2cDevice.WriteByte((byte)reg);
+            var buf = new byte[2];
+            _i2cDevice.Read(buf);
+
+            return (ushort)(buf[0] << 8 | buf[1]);
```

Comment:
Is this endianess going to be correct on all machines? Usually we suggest to instead of manually doing this, to call BinaryPrimitives.ReadInt16BigEndian or BinaryPrimitives.ReadInt16LittleEndian which will handle that on all precessors for you.

Reply:
It's not related to the machine's endianess, it's how the MCP9808 is sending it, high byte first, then low byte.